Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -7117,7 +7117,11 @@ def is_integration_command_execution():
     try:
         return demisto.callingContext['context']['ExecutedCommands'][0]['moduleBrand'] != 'Scripts'
     except (KeyError, IndexError, TypeError):
-        return True
+        try:
+            # In dynamic-section scripts ExecutedCommands is None and another way is needed to verify if we are in a Script.
+            return demisto.callingContext['context']['ScriptName'] == ''
```

Comment:
In dynamic sections, the check before raises an exception since ExecutedCommands is None, and we assume it is an integration. It seems that usually when running any scripts (dynamic-sections or not) the `context.ScriptName` parameter is set. But I think that running an integration command can also result sometimes in a context.ScriptName, therefore I rather leave the previous check as well.